src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildData.java

```diff
@@ -236,6 +225,59 @@ public BuildData(Run run, TaskListener listener) throws IOException, Interrupted
         }
     }
 
+    private static String getBaseJobName(Run run, EnvVars envVars) {
+        Job<?, ?> job = run.getParent();
+        if (job != null) {
+            ItemGroup<?> jobParent = job.getParent();
+            if (jobParent != null) {
+                try {
+                    String jobParentFullName = jobParent.getFullName();
+                    if (StringUtils.isNotBlank(jobParentFullName)) {
+                        return jobParentFullName;
+                    }
+                } catch (NullPointerException e) {
+                    // this is possible if data is not mocked properly in unit tests
+                }
+            }
+        }
+        String jobName = job.getName();
+        if (StringUtils.isNotBlank(jobName)) {
+            return jobName;
+        }
+        if (envVars != null) {
+            String envJobBaseName = envVars.get("JOB_BASE_NAME");
+            if (StringUtils.isNotBlank(envJobBaseName)) {
+                return envJobBaseName;
+            }
+            String envJobName = envVars.get("JOB_NAME");
+            if (StringUtils.isNotBlank(envJobName)) {
+                return envJobName;
+            }
+        }
+        return "unknown";
```

Comment:
For the `JOB_BASE_NAME`, wouldn't better to check if the env var is set, and if not, use the `getJobName` functions?
```suggestion
if (envVars != null) {
    String envJobBaseName = envVars.get("JOB_BASE_NAME");
    if (StringUtils.isNotBlank(envJobBaseName)) {
        return envJobBaseName;
    }
}
              
return getJobName(run, envVars);
```

Reply:
I was afraid to change the logic too much, since I didn't know what the difference was between the API and the environment variables. That is why I added using the latter only as a fallback - to minimise the changes in behaviour.

If we can be sure that the env var never contradicts whatever is reported by the API, then surely we can change it.
What do you think?

Reply:
I would say that the API and env var **should** be aligned, but I don't have enough context of the Jenkins core implementation to put my finger on it 😅.

Reply:
Judging by the core code it does seem to be the case. Still I think we should err on the side of caution.

I have changed `getBaseJobName`  to delegate to `getJobName` similarly to what you suggested, but still left the API as the higher priority option, so the behaviour is the following:

1. Try to get the base name from the API
2. Fall back to `JOB_BASE_NAME`
3. Fall back to `getJobName`